File: backend/app/models/task.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

TaskType = Literal["download", "convert", "separate", "remove_bg"]
TaskStatus = Literal["queued", "processing", "done", "error", "cancelled"]


class Task(BaseModel):
    id: str
    type: TaskType
    name: str
    status: TaskStatus
    progress: int = Field(default=0, ge=0, le=100)
    stage: str | None = None
    output_files: list[str] = Field(default_factory=list)
    error: str | None = None
    size: str | None = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/task.ts
import type { TaskStatus, TaskType } from '../lib/task-types'

export interface Task {
  id: string
  type: TaskType
  name: string
  status: TaskStatus
  progress: number
  stage?: string | null
  output_files: string[]
  error?: string | null
  size?: string | null
  created_at: string
  updated_at: string
}

export interface TasksResponse {
  tasks: Task[]
}

export interface TaskStartResponse {
  task_id: string
  type: TaskType
  status: 'started'
}
